components/camel-xj/src/main/java/org/apache/camel/component/xj/XmlJsonStreamWriter.java

```diff
@@ -0,0 +1,640 @@
+                        }
+                    } else if (childs.size() == 1 && childs.get(0).xmlEvent == XMLEvent.CHARACTERS) {
+                        // just character childs.
+
+                        // empty root element
+                        if (this.parent.jsonToken == JsonToken.NOT_AVAILABLE) {
+                            jsonToken = JsonToken.START_OBJECT;
+
+                            final TreeElement child = childs.get(0);
+                            if (isWhitespace(child.value)) {
+                                childs.remove(0);
+                            } else {
+                                // create new intermediary element
+                                final TreeElement treeElement = new TreeElement(this, -1, JsonToken.FIELD_NAME, JSON_MIXED_CONTENT_TEXT_KEY);
+                                treeElement.addChild(child);
+                                childs.set(childs.indexOf(child), treeElement);
+                                child.parent = treeElement;
+                            }
+                        } else {
+                            jsonToken = JsonToken.FIELD_NAME;
+                        }
+                    } else {
+                        // mixed content fixup.
+                        final Iterator<TreeElement> iterator = childs.iterator();
+                        while (iterator.hasNext()) {
+                            TreeElement element = iterator.next();
+                            if (element.jsonToken == JsonToken.VALUE_STRING) {
+                                if (isWhitespace(element.value)) {
+                                    // remove element if is (ignorable-) whitespace
+                                    iterator.remove();
+                                } else {
+                                    // create new intermediary element
+                                    final TreeElement treeElement = new TreeElement(this, -1, JsonToken.FIELD_NAME,
+                                            element.name != null ? element.name : JSON_MIXED_CONTENT_TEXT_KEY);
+                                    treeElement.addChild(element);
+                                    childs.set(childs.indexOf(element), treeElement);
+                                    element.parent = treeElement;
+                                    element.jsonToken = JsonToken.VALUE_STRING;
+                                }
+                            }
+                        }
+
+                        jsonToken = JsonToken.START_OBJECT;
+
+                        final Map<String, Set<TreeElement>> childElementsMap = childs.stream()
+                                .collect(Collectors.groupingBy(o -> o.name, HashMap::new, Collectors.toCollection(LinkedHashSet::new)));
+
+                        // create arrays if element with the same name occurs more than once.
+                        for (Map.Entry<String, Set<TreeElement>> mapEntry : childElementsMap.entrySet()) {
```

Comment:
Lines 382-403 are exactly same as lines 457-479.  And the 511-524 are pretty much similar. Is it possible to refactor to single method?

Reply:
done.